### PR TITLE
Partially implement `_CFXPCCreateCFObjectFromXPCObject`

### PR DESCRIPTION
**Pull request by facekapow**
Title says it all.

### CODE REVIEW

File: CFXPCBridge.c [hunk 1]
```diff
@@ -29,14 +29,86 @@
...
-	// CF counterparts.
+#include <xpc/xpc.h>
+
+CFTypeRef _CFXPCCreateCFObjectFromXPCObject(xpc_object_t xo) {
+	xpc_type_t type = xpc_get_type(xo);
+
+	if (type == XPC_TYPE_DICTIONARY) {
+		size_t count = xpc_dictionary_get_count(xo);
+		__block CFStringRef* keys = malloc(count * sizeof(CFStringRef));
+		__block CFTypeRef* values = malloc(count * sizeof(CFTypeRef));
```
Comment (bugaevc) [inline CFXPCBridge.c]: I don't think the pointers have to be `__block`

File: CFXPCBridge.c [hunk 2]
```diff
@@ -29,14 +29,86 @@
...
+		return array;
+	}
+
+	if (type == XPC_TYPE_STRING) {
+		return CFStringCreateWithCString(NULL, xpc_string_get_string_ptr(xo), kCFStringEncodingUTF8);
+	}
+
+	if (type == XPC_TYPE_BOOL) {
+		CFBooleanRef boolean = xpc_bool_get_value(xo) == true ? kCFBooleanTrue : kCFBooleanFalse;
+		CFRetain(boolean); // are we supposed to do this? ¯\_(ツ)_/¯
```
Comment (bugaevc) [inline CFXPCBridge.c]: Yes (but it should be a no-op).

File: CFXPCBridge.c [hunk 3]
```diff
@@ -29,14 +29,86 @@
...
+		return boolean;
+	}
+
+	if (type == XPC_TYPE_DATA) {
+		return CFDataCreate(NULL, xpc_data_get_bytes_ptr(xo), xpc_data_get_length(xo));
+	}
+
+	if (type == XPC_TYPE_INT64) {
+		int64_t val = xpc_int64_get_value(xo);
+	}
```
Comment (bugaevc) [inline CFXPCBridge.c:95]: Missing return

File: CFXPCBridge.c [hunk 4]
```diff
@@ -29,14 +29,86 @@
...
+
+	if (type == XPC_TYPE_UINT64) {
+		uint64_t val = xpc_uint64_get_value(xo);
+		return CFNumberCreate(NULL, kCFNumberSInt64Type, &val);
+	}
+
+	if (type == XPC_TYPE_NULL) {
+		CFRetain(kCFNull); // again, do we have to retain constants?
+		return kCFNull;
```
Comment (bugaevc) [inline CFXPCBridge.c]: You can even write it in one line, like in objc: `return CFRetain(kCFNull);`

**Comment by facekapow** [comment]
Removed unnecessary `__block`s, simplified `CFRetain` code, and implemented `int64` XPC -> CF conversion (`uint64` is almost exactly the same code, so when I implemented it I thought I had implemented `int64`).

**Pull request merged**